rclc/test/rclc/test_executor_handle.cpp

```diff
@@ -16,6 +16,16 @@
 #include <gtest/gtest.h>
 #include "rclc/executor_handle.h"
 
+TEST(Test, executor_handle_reduced_type) {
+  rclc_executor_handle_type_t full_type;
+  rclc_executor_handle_type_t reduced_type;
+  for (int i = -10; i < 30; i++) {
+    full_type = (rclc_executor_handle_type_t) i;
+    reduced_type = rclc_executor_handle_reduced_type(full_type);
+    EXPECT_EQ(reduced_type, rclc_executor_handle_reduced_type(reduced_type)) <<
+      "idempotency violation";
+  }
+}
```

Comment:
I didn't want to put in an exhaustive test because an exhaustive test would look complete but (being a re-implementaton) would pass on all of the most likely failure cases, especially New types.
codecov only flagged the `default` case as unexplored, and strict idempotency is that case's strongest property.

I'll add some more cases and explain that better in the test code.